content/cookbook/transactions/add-priority-fees.md

```diff
@@ -29,6 +29,121 @@ compute budget. The value provided will replace the default value. Transactions
 should request the minimum amount of CU required for execution to maximize
 throughput, or minimize fees.
 
+<Tabs groupId="language" items={['web3.js v2', 'web3.js v1']}>
+
+<Tab value="web3.js v2">
+
+```typescript filename="add-memo.ts" {61-72} {37-38} {77-87}
+import {
+  airdropFactory,
+  appendTransactionMessageInstructions,
+  createSolanaRpc,
+  createSolanaRpcSubscriptions,
+  createTransactionMessage,
+  devnet,
+  generateKeyPairSigner,
+  getComputeUnitEstimateForTransactionMessageFactory,
+  getSignatureFromTransaction,
+  lamports,
+  pipe,
+  prependTransactionMessageInstructions,
+  sendAndConfirmTransactionFactory,
+  setTransactionMessageFeePayerSigner,
+  setTransactionMessageLifetimeUsingBlockhash,
+  signTransactionMessageWithSigners,
+} from "@solana/web3.js";
+import {
+  getSetComputeUnitLimitInstruction,
+  getSetComputeUnitPriceInstruction,
+} from "@solana-program/compute-budget";
+import { getAddMemoInstruction } from "@solana-program/memo";
+
+async function writeMemoWithPriorityFees(message: string) {
+  // Create an RPC.
+  const CLUSTER = "devnet";
+  const rpc = createSolanaRpc(devnet(`https://api.${CLUSTER}.solana.com`));
+  const rpcSubscriptions = createSolanaRpcSubscriptions(
+    devnet(`wss://api.${CLUSTER}.solana.com`),
+  );
+
+  // Create an airdrop function.
+  const airdrop = airdropFactory({ rpc, rpcSubscriptions });
+
+  // Create a utility that estimates a transaction message's compute consumption.
+  const getComputeUnitEstimate =
+    getComputeUnitEstimateForTransactionMessageFactory({ rpc });
+
+  // Create a transaction sending function.
+  const sendAndConfirmTransaction = sendAndConfirmTransactionFactory({
+    rpc,
+    rpcSubscriptions,
+  });
+
+  // Create and fund an account.
+  const keypairSigner = await generateKeyPairSigner();
+  console.log("Created an account with address", keypairSigner.address);
+  console.log("Requesting airdrop");
+  await airdrop({
+    commitment: "confirmed",
+    lamports: lamports(1000_000n),
+    recipientAddress: keypairSigner.address,
+  });
+  console.log("Airdrop confirmed");
+
+  // Create a memo transaction.
+  console.log("Creating a memo transaction");
+  const { value: latestBlockhash } = await rpc.getLatestBlockhash().send();
+  const transactionMessage = pipe(
+    createTransactionMessage({ version: "legacy" }),
+    m => setTransactionMessageFeePayerSigner(keypairSigner, m),
+    m => setTransactionMessageLifetimeUsingBlockhash(latestBlockhash, m),
+    m =>
+      appendTransactionMessageInstructions(
+        [
+          getSetComputeUnitPriceInstruction({ microLamports: 5000n }),
+          getAddMemoInstruction({ memo: message }),
+        ],
+        m,
+      ),
+  );
+
+  // Figure out how many compute units to budget for this transaction
+  // so that you can right-size the compute budget to maximize the
+  // chance that it will be selected for inclusion into a block.
```

Comment:
since the real compute consumed could vary when it hits the network with more complicated programs, do you think its worth adding in some extra cus? iirc, the `helpers` library adds in some small amount of cus to add a little wiggle room (since that is what star atlas found to be effective)

Reply:
Hmm do you know what staratlas is doing there? Sounds pretty random to just add some number. The Helpers just simulate with a 1.4 mil cu. I was hoping the new function works better. At least this is what luscher recommended. Ill ask him

Reply:
https://github.com/solana-labs/solana-web3.js/tree/master/packages/library#getcomputeunitestimatefortransactionmessagefactoryrpc Ah there is this. Still not sure what value i should recommend. Some random const, some random multiplier? Maybe just a link to the docs.

Reply:
I think we add a note that its an estimate but a common practice is to add some buffer amount, like 5-10% more. Maybe add a Callout via a markdown quoteblock (i.e. `> Example it here`)